cola/widgets/startup.py

```diff
@@ -265,16 +336,397 @@ def handle_broken_repo(self, index):
             self.context.settings.remove_recent(repo['path'])
             self.context.settings.save()
 
-            item = self.bookmarks_model.item(index.row())
-            self.items.remove(item)
-            self.bookmarks_model.removeRow(index.row())
+            item = model.item(index.row())
+            items.remove(item)
+            model.removeRow(index.row())
+
+            other_model_row_count = other_model.rowCount()
+            for i in range(other_model_row_count):
+                other_index = other_model.index(i, 0)
+                other_item = other_model.itemFromIndex(other_index)
+
+                if (
+                    isinstance(other_item, PromptWidgetItem)
+                    and other_item.path == repo['path']
+                ):
+                    other_items.remove(other_item)
+                    other_model.removeRow(other_index.row())
+                    return
 
     def get_selected_bookmark(self):
         selected = self.bookmarks.selectedIndexes()
+        if not selected:
+            selected = self.recent.selectedIndexes()
+
         if selected and selected[0].row() != 0:
             return self.bookmarks_model.data(selected[0], Qt.UserRole)
         return None
 
+    def set_model(self, items, mode):
+        if mode == LIST_VIEW_BOOKMARK:
+            model = self.bookmarks_model
+            self.bookmark_items = new_items = []
+        elif mode == LIST_VIEW_RECENT:
+            model = self.recent_model
+            self.recent_items = new_items = []
+        else:
+            return
```

Comment:
Is the `else:` ever reached, or is it unreachable?

This is another example where the internal complexity might be able to be reduced. The `mode` is basically deciding which of the internal two models to use. That seems like it can be improved by factoring that stuff out.

Reply:
This `else:` never reached and for completeness too, and it might be redundant.
OK, I'll eliminate it.